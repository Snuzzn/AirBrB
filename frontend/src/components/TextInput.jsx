import React from 'react';
import PropTypes from 'prop-types';

const TextInput = ({ label, type, placeholder, setState }) => {
  return (
    <>
      <input
        aria-label={label}
        className="bg-gray-100 focus:bg-transparent w-full text-md block mt-2 p-2 border rounded-lg focus:outline-none focus:ring focus:ring-red-300"
        type={type}
        placeholder={placeholder}
        onChange={(e) => setState(e.target.value)}
      />
    </>
  )
}

export default TextInput;

TextInput.propTypes = {
  label: PropTypes.string,
  type: PropTypes.string,
  placeholder: PropTypes.string,
  setState: PropTypes.func
}
