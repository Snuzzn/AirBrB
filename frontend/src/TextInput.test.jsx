import React from 'react';
import { render, screen } from '@testing-library/react';
import TextInput from './components/TextInput';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

describe('TextInput', () => {
  /**
   * TextInput captures user input in login and register. It takes
   * in an onChange function, the type of input, an optional placeholder,
   * and a custom aria-label for accessibility.
   */

  const noop = () => {};

  it('triggers onChange event on user input', () => {
    const onChange = jest.fn();
    const inputField = shallow(<TextInput
        type='text'
        setState={onChange}
    />);
    const event = { target: { value: 'a' } }
    inputField.simulate('change', event);
    expect(onChange).toHaveBeenCalledTimes(1);
  })

  it('captures user input on change', () => {
    const onChange = jest.fn();
    const inputField = shallow(<TextInput
        type='text'
        setState={onChange}
    />);
    const event = { target: { value: 'testing this value' } }
    inputField.simulate('change', event);
    expect(onChange).toHaveBeenCalledWith('testing this value');
  })

  it('aria-label is defined with prop value', () => {
    const inputField = shallow(<TextInput
      label='Test Label!'
      setState={noop}
    />);
    expect(inputField.prop('aria-label')).toBe('Test Label!');
  })

  it('type is defined with prop value', () => {
    const inputField = shallow(<TextInput
      type='password'
      setState={noop}
    />);
    expect(inputField.prop('type')).toBe('password');
  })

  it('placeholder is defined with prop value', () => {
    const inputField = shallow(<TextInput
      type='text'
      placeholder='Test Placeholder!'
      setState={noop}
    />);
    expect(inputField.prop('placeholder')).toBe('Test Placeholder!');
  })

  it('uses placeholder text defined in prop', () => {
    render(<TextInput
      type='text'
      placeholder='Placeholder!'
      setState={noop}
    />);
    expect(screen.getByPlaceholderText('Placeholder!')).toBeInTheDocument();
  })

  it('uses default placeholder text if no prop passed in', () => {
    render(<TextInput
      type='text'
      setState={noop}
    />);
    // TextInput's default placeholder is 'Please enter details'
    expect(screen.getByPlaceholderText('Please enter details')).toBeInTheDocument();
  })

  // Snapshots
  it('renders with minimal props', () => {
    const inputField = renderer.create(<TextInput onChange={noop} />).toJSON();
    expect(inputField).toMatchSnapshot();
  })

  it('renders with provided placeholder and type', () => {
    const inputField = renderer.create(<TextInput
      onChange={noop}
      type='text'
      placeholder='placeholder test'
    />).toJSON();
    expect(inputField).toMatchSnapshot();
  })
})
