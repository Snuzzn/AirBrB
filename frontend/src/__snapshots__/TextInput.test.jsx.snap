// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput renders with minimal props 1`] = `
<input
  className="bg-gray-100 focus:bg-transparent w-full text-md block mt-2 p-2 border rounded-lg focus:outline-none focus:ring focus:ring-red-300"
  onChange={[Function]}
  placeholder="Please enter details"
/>
`;

exports[`TextInput renders with provided placeholder and type 1`] = `
<input
  className="bg-gray-100 focus:bg-transparent w-full text-md block mt-2 p-2 border rounded-lg focus:outline-none focus:ring focus:ring-red-300"
  onChange={[Function]}
  placeholder="placeholder test"
  type="text"
/>
`;
